bokeh/models/mappers.py

```diff
@@ -63,3 +65,229 @@ class LinearColorMapper(ColorMapper):
+
+        if palette is not None:
+            tmp_palette = expandPalette(palette)
+            palette = tmp_palette['palette']
+            alpha_hold = tmp_palette['alpha_hold']
+            kwargs['palette'] = palette
+
+        if alpha is not None and palette is not None:
+            if not isinstance(alpha, list):
+                # Convert the scalar object to a vector so that the Seq property
+                # type can be used.
+                alpha = [alpha] * len(palette)
+
+            # Overwrite the alpha values with those specified by the colors in the palette.
+            # This may not be the desired functionality.  Who should win, the information in
+            # the palette, or the direct alpha information specified in the alpha parameter?
+            for k,v in list(alpha_hold.items()):
+                alpha[k] = v
+
+            # Check to ensure that all of the alpha values are ok.
+            if any([x > 1 for x in alpha]):
+                raise ValueError('alpha can not have any values greater than 1.0')
+
+            if any([x < 0 for x in alpha]):
+                raise ValueError('alpha cannot have any values less than 0.0')
+
+            kwargs['alpha'] = alpha
+
+        if segments is not None: kwargs['segments'] = segments
+
+        kwargs['interpolationMethod'] = interpolationMethod
+
+        super(SegmentedColorMapper, self).__init__(**kwargs)
+
+
+class LinearColorMapperReplacement(SegmentedColorMapper):
+
+    low = Float(help="""
+    The minimum value of the range to map into the palette. Values below
+    this are clamped to ``low``.
+    """)
+
+    high = Float(help="""
+    The maximum value of the range to map into the palette. Values above
+    this are clamped to ``high``.
+    """)
+
+    # TODO: (jc) what is the color code for transparent?
+    # TODO: (bev) better docstring
+    reserve_color = Color("#ffffff", help="""
+    Used by Abstract Rendering.
+    """)
+
+    # TODO: (bev) better docstring
+    reserve_val = Float(default=None, help="""
+    Used by Abstract Rendering.
+    """)
+
+    def __init__(self, palette=None, low = None, high = None, alpha=1, reserve_color = None, reserve_val = None, **kwargs):
+
+        if palette is not None:
+            tmp_palette = expandPalette(palette)
+            palette = tmp_palette['palette']
+
+            if not isinstance(palette, list):
+                raise TypeError('the palette failed to expand to an array')
+
+            # Duplicate the final color to mimic the behavior of the original
+            # LinearColorMapper Class
+            palette.append(palette[len(palette) - 1])
+
+        # Reuse the expandPalette framework to convert the single reserve color
+        # to a hex code which can be passed to the JS code.  (This feels a bit hackinsh
+        # but it is better than simply duplicating the code below.)
+        if reserve_color is not None:
+            reserve_color = expandPalette([reserve_color])['palette'][0]
```

Comment:
should also be unneeded. `reserve_color` is a holdover from abstract rendering, which has been removed from Bokeh proper. We might want high/low colors to clamp out of range values to, but those should get better names. e.g. `high_color`, `low_color`